BlockchainSdk/Blockchains/Solana/SolanaWalletManager.swift

```diff
@@ -180,53 +177,54 @@ extension SolanaWalletManager: TransactionSender {
 private extension SolanaWalletManager {
     /// Combine `accountCreationFeePublisher`, `accountExistsPublisher` and `minimalBalanceForRentExemption`
     func destinationAccountInfo(destination: String, amount: Amount) -> AnyPublisher<DestinationAccountInfo, Error> {
-        let accountCreationFeePublisher = accountCreationFeePublisher(amount: amount)
         let accountExistsPublisher = accountExists(destination: destination, amountType: amount.type)
         let rentExemptionBalancePublisher = networkService.minimalBalanceForRentExemption()
-
-        
-        /*
-         1. we get our account (if not, we return the error if yes, we go further)
-         2. we get the account of the recipient of the token, if it is fee = 0,
-         3. if it is not, then we take the space of our account and receive a fee for creation
-        */
         
-        return Publishers.Zip3(accountCreationFeePublisher, accountExistsPublisher, rentExemptionBalancePublisher)
-            .map { accountCreationFee, accountExists, rentExemption in
-                let creationFee: Decimal
-                if accountExists {
-                    creationFee = 0
-                } else if amount.type == .coin && amount.value >= rentExemption {
-                    creationFee = 0
+        return Publishers.Zip(accountExistsPublisher, rentExemptionBalancePublisher)
+            .withWeakCaptureOf(self)
+            .flatMap { (manager, values) in
+                let accountExistsInfo = values.0
+                let rentExemption = values.1
```

Comment:
nit просто на будущее )

```suggestion
                let (accountExistsInfo, rentExemption) = values
```